core/src/apresolve.rs

```diff
@@ -42,51 +81,60 @@ async fn try_apresolve(
     let body = hyper::body::to_bytes(response.into_body()).await?;
     let data: ApResolveData = serde_json::from_slice(body.as_ref())?;
 
-    let mut aps = data.ap_list.into_iter().filter_map(|ap| {
-        let mut split = ap.rsplitn(2, ':');
-        let port = split
-            .next()
-            .expect("rsplitn should not return empty iterator");
-        let host = split.next()?.to_owned();
-        let port: u16 = port.parse().ok()?;
-        Some((host, port))
-    });
-    let ap = if ap_port.is_some() || proxy.is_some() {
-        aps.find(|(_, p)| *p == port)
-    } else {
-        aps.next()
-    }
-    .ok_or("no valid AP in list")?;
-
-    Ok(ap)
+    Ok(data)
 }
 
-pub async fn apresolve(proxy: Option<&Url>, ap_port: Option<u16>) -> (String, u16) {
-    try_apresolve(proxy, ap_port).await.unwrap_or_else(|e| {
-        warn!("Failed to resolve Access Point: {}, using fallback.", e);
-        (String::from(AP_FALLBACK), 443)
-    })
+pub async fn apresolve(proxy: Option<&Url>, ap_port: Option<u16>) -> AccessPoints {
+    let data = try_apresolve(proxy).await.unwrap_or_else(|e| {
+        warn!("Failed to resolve access points: {}, using fallbacks.", e);
+        ApResolveData {
+            accesspoint: vec![],
+            dealer: vec![],
+            spclient: vec![],
+        }
```

Comment:
Maybe `derive(Default)` to make it shorter?

Reply:
Sure thing, done in 510c874.